src/mame/drivers/dynax.cpp

```diff
@@ -4909,6 +4923,12 @@ void dynax_state::mjreach(machine_config &config)
 	m_mainlatch->q_out_cb<0>().set(FUNC(dynax_state::flipscreen_w)); // not inverted
 }
 
+void dynax_state::ougonhai(machine_config &config)
+{
+	tenkai(config);
+	tmp91640_device* tmp = reinterpret_cast<tmp91640_device *>(m_maincpu.target());
```

Comment:
Please use `downcast<...>(*m_maincpu)` – a `reinterpret_cast` may not do the right thing.